pkg/arvo/app/group-store.hoon

```diff
@@ -33,50 +69,59 @@
       gc          ~(. group-core bowl)
       def         ~(. (default-agent this %|) bowl)
   ::
-  ++  on-init            on-init:def
+  ++  on-init  on-init:def
   ++  on-save   !>(state)
   ++  on-load
-    |=  old=vase
-    `this(state !<(state-zero old))
+    |=  =old=vase
+    =/  old  !<(versioned-state old-vase)
+    ?.  ?=(%1 -.old)
+      `this
+    `this(state old)
   ::
   ++  on-poke
     |=  [=mark =vase]
     ^-  (quip card _this)
     ?>  (team:title our.bowl src.bowl)
     =^  cards  state
-      ?:  ?=(%group-action mark)
-        (poke-group-action:gc !<(group-action vase))
+      ?:  ?=(?(%group-update %group-action) mark)
+        (poke-group-update:gc !<(update:store vase))
       (on-poke:def mark vase)
     [cards this]
   ::
   ++  on-watch
     |=  =path
     ^-  (quip card _this)
     ?>  (team:title our.bowl src.bowl)
-    |^
-    =/  cards=(list card)
-      ?+    path  (on-watch:def path)
-          [%all ~]   (give %group-initial !>(groups))
-          [%updates ~]  ~
-          [%keys ~]  (give %group-update !>([%keys ~(key by groups)]))
-          [%group *]
-        (give %group-update !>([%path (~(got by groups) t.path) t.path]))
-      ==
-    [cards this]
-    ::
-    ++  give
-      |=  =cage
-      ^-  (list card)
-      [%give %fact ~ cage]~
-    --
+    ?>  ?=([%groups ~] path)
+    :_  this
+    [%give %fact ~ %group-update !>([%initial groups])]~
   ::
   ++  on-leave  on-leave:def
   ::
   ++  on-peek
     |=  =path
     ^-  (unit (unit cage))
     ?+  path  (on-peek:def path)
-      [%x *]  ``noun+!>((~(get by groups) t.path))
+        [%x %groups @ @ ~]
+      =/  group-id
+        (group-id:de-path:store t.t.path)
+      ?~  group-id  ~
```

Comment:
(I always trip myself up on this but) I think `~` is for "no answer", whereas `[~ ~]` is for "invalid request"/"doesn't exist", which I'm pretty sure what you intend to say here.